inst/shiny/modules/tab_visuals.R

```diff
@@ -0,0 +1,560 @@
+    })
+
+    # Reactive expression for summary table based on selected group and parameters
+    summary_stats <- reactive({
+      req(input$summarygroupby, input$paramselect)
+
+      # Calculate summary stats and filter by selected parameters
+      calculate_summary_stats(res_nca(), input$summarygroupby) %>%
+        filter(PPTESTCD %in% input$paramselect)  %>%
+        rename(PARAM = PPTESTCD)
+    })
+
+    # render the reactive summary table in a data table
+    output$descriptivestats <- DT::renderDataTable({
+      req(summary_stats())
+      DT::datatable(
+        data = summary_stats(),
+        extensions = "FixedHeader",
+        options = list(
+          scrollX = TRUE,
+          scrollY = TRUE,
+          lengthMenu = list(c(10, 25, -1), c("10", "25", "All")),
+          fixedHeader = TRUE
+        )
+      )
+    })
+
+    output$descriptivestats2 <- DT::renderDataTable({
+      req(res_nca())
+      DT::datatable(
+        data = calculate_summary_stats(res_nca())  %>% rename(PARAM = PPTESTCD),
+        extensions = "FixedHeader",
+        options = list(
+          scrollX = TRUE,
+          scrollY = TRUE,
+          lengthMenu = list(c(10, 25, -1), c("10", "25", "All")),
+          fixedHeader = TRUE
+        )
+      )
+    })
+
+    # Save summary stats to improve
+    observeEvent(input$downloadsum, {
+      showModal(modalDialog(
+        title = "Please enter the path to the folder on improve for your results:",
+        textInput(ns("pathresults"), "Path:"),
+        textInput(ns("filename"), "Enter filename of choice:", ""),
+        actionButton(ns("go2"), "GO"),
+        footer = modalButton("Close")
+      ))
+    })
+
+    # alternatively: save locally
+    output$downloadsum_browser <- downloadHandler(
+      filename = function() {
+        paste("NCA_summary.csv", sep = "_")
+      },
+      content = function(file) {
+        write.csv(summary_stats(), file)
+      }
+    )
+
+    # TAB: Parameter Box plots ----------------------------------------------------
+
+    # Create formatted Box plot data: PKNCA + PP results, linking DOSEA + PPTESTCD
+    boxplotdata <- reactive({
+      group_columns <- unname(unlist(res_nca()$data$conc$columns$groups))
+
+      left_join(
+        res_nca()$result %>%
+          filter(
+            end == Inf | startsWith(PPTESTCD, "aucint")
+          ),
+        res_nca()$data$conc$data %>%
+          distinct(across(all_of(group_columns)), .keep_all = TRUE),
+        by = group_columns,
+        keep = FALSE
+      ) %>%
+        # Intervals should also be considered as differentiated options each
+        mutate(
+          PPTESTCD = ifelse(
+            startsWith(PPTESTCD, "aucint"),
+            paste0(PPTESTCD, "_", start, "-", end),
+            PPTESTCD
+          )
+        )
+    })
+
+    # select which parameter to box or violin plot
+    output$selectboxplot <- renderUI({
+      param_choices <- boxplotdata()$PPTESTCD %>% unique()
```

Comment:
issue: Please avoid mixing base R and dplyr patterns.
```suggestion
      param_choices <- boxplotdata() %>%
        pull(PPTESTCD) %>%
        unique()
```